py-polars/polars/functions.py

```diff
@@ -216,23 +216,51 @@ def read_csv(
 
         return from_arrow(tbl, rechunk)
 
-    df = DataFrame.read_csv(
-        file=file,
-        infer_schema_length=infer_schema_length,
-        batch_size=batch_size,
-        has_headers=has_headers,
-        ignore_errors=ignore_errors,
-        stop_after_n_rows=stop_after_n_rows,
-        skip_rows=skip_rows,
-        projection=projection,
-        sep=sep,
-        columns=columns,
-        rechunk=rechunk,
-        encoding=encoding,
-        n_threads=n_threads,
-        dtype=dtype,
-        low_memory=low_memory,
-    )
+    if isinstance(file, str) and file.endswith(".gz"):
+        try:
+            # Try to use python-isal for faster gzip decompresssion.
+            from isal import igzip as gzip_mod
+        except ImportError:
+            # Use normal gzip module, if python-isal is not installed.
+            import gzip as gzip_mod
+
+        with gzip_mod.open(file, "rb") as fh:
```

Comment:
Maybe we can define a closure or a lambda above that takes only the new file (`fh` or `file`) as argument and captures the other args from the scope.


```python
def read_df(file):
    return DataFrame.read_csv(
	file=file,
	infer_schema_length=infer_schema_length,
	batch_size=batch_size,
	has_headers=has_headers,
	ignore_errors=ignore_errors,
	stop_after_n_rows=stop_after_n_rows,
	skip_rows=skip_rows,
	projection=projection,
	sep=sep,
	columns=columns,
	rechunk=rechunk,
	encoding=encoding,
	n_threads=n_threads,
	dtype=dtype,
	low_memory=low_memory,
        )


if ...:
     with gzip_mod.open(file, "rb") as fh:
          df = read_df(fh)
else:
    df = read_df(file)

```

Reply:
So to make read_df a subfunction in read_csv?

Reply:
Yes, but then I am less likely to make mistakes when I add or remove some arguments.